packages/desktop/electron/preload.ts

```diff
@@ -0,0 +1,43 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { ipcRenderer, contextBridge } from 'electron';
+import { WriteFileOptions } from 'fs';
+
+import * as eventNames from '../ipcEventNames';
+import { WALWindowAPI } from '../renderer';
+
+const api: WALWindowAPI = {
+  getPlatform: () => ipcRenderer.sendSync(eventNames.IPC_GET_PLATFORM_SYNC),
+  joinDirPath: (args: string[]) => ipcRenderer.invoke(eventNames.IPC_CALL_PATH_JOIN, args),
+
+  fsCloseSync: (fd: number) => ipcRenderer.invoke(eventNames.IPC_FS_CLOSE_SYNC, fd),
```

Comment:
wait... i might have misunderstood something somewhere. I thought the point of having this preload script is so that node.js stuff can be done here. isn't this function supposed to just call fs.closeAsync directly? 

orthogonal to that, I feel we shouldn't expose this kind of low level APIs through the bridge, because it can defeat the purpose of having an isolation. A more secure approach would be to expose an API that's "readRecentCombatLogs()" or something like that which narrows down what's possible. What are your thoughts?